### PR TITLE
Add new settings for remote publication

### LINKED ISSUE #8168: [DOC] Modify documentation for remote cluster state publication
**What do you want to do?**
 
- [x] Request a change to existing documentation
- [ ] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request.
New cluster settings are being added for remote cluster state publication.
We need to document these settings.

***Version:** List the OpenSearch version to which this issue applies, e.g. 2.14, 2.12--2.14, or all.
2.17
 
**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.
https://github.com/opensearch-project/OpenSearch/pull/15575
https://github.com/opensearch-project/OpenSearch/pull/15622
https://github.com/opensearch-project/OpenSearch/pull/15478

### PR DESCRIPTION
**Pull request by soosinha**
### Description
New settings are being added for remote cluster state publication. 

### Issues Resolved
resolves #8168

### Version
2.17

### Frontend features
NA

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _tuning-your-cluster/availability-and-recovery/remote-store/remote-cluster-state.md [hunk 1]
```diff
@@ -89,8 +93,13 @@ You do not have to use different remote store repositories for state and routing
...
 
-Setting | Default | Description
-:--- | :--- | :---
-`cluster.remote_store.state.read_timeout` | 20s | The amount of time to wait for remote state download to complete on the follower node.
-`cluster.remote_store.routing_table.path_type` | HASHED_PREFIX | The path type to be used for creating an index routing path in the blob store. Valid values are `FIXED`, `HASHED_PREFIX`, and `HASHED_INFIX`.
-`cluster.remote_store.routing_table.path_hash_algo` | FNV_1A_BASE64 | The algorithm to be used for constructing the prefix or infix of the blob store path. This setting is applied if `cluster.remote_store.routing_table.path_type` is `hashed_prefix` or `hashed_infix`. Valid algorithm values are `FNV_1A_BASE64` and `FNV_1A_COMPOSITE_1`.
+Setting | Default  | Description
+:--- |:---| :---
+`cluster.remote_store.state.read_timeout` | 20s | The amount of time to wait for the remote state download to complete on the follower node.
+`cluster.remote_store.state.path.prefix` | "" (Empty string) | The fixed prefix to add for the index metadata files in the blob store.
```
Comment (natebower) [inline _tuning-your-cluster/availability-and-recovery/remote-store/remote-cluster-state.md:99]: "to add to"?

File: _tuning-your-cluster/availability-and-recovery/remote-store/remote-cluster-state.md [hunk 2]
```diff
@@ -89,8 +93,13 @@ You do not have to use different remote store repositories for state and routing
...
+`cluster.remote_store.index_metadata.path_type` | `HASHED_PREFIX`  | The path type used for creating an index metadata path in the blob store. Valid values are `FIXED`, `HASHED_PREFIX`, and `HASHED_INFIX`.
+`cluster.remote_store.index_metadata.path_hash_algo` | `FNV_1A_BASE64 `  | The algorithm that constructs the prefix or infix for the index metadata path in the blob store. This setting is applied if the ``cluster.remote_store.index_metadata.path_type` setting is `hashed_prefix` or `hashed_infix`. Valid algorithm values are `FNV_1A_BASE64` and `FNV_1A_COMPOSITE_1`.
```
Comment (natebower) [inline _tuning-your-cluster/availability-and-recovery/remote-store/remote-cluster-state.md:101]: Are the lowercase `hashed_prefix` and `hashed_infix` here the same as the all caps ones on the preceding line? If so, they should match.

File: _tuning-your-cluster/availability-and-recovery/remote-store/remote-cluster-state.md [hunk 3]
```diff
@@ -89,8 +93,13 @@ You do not have to use different remote store repositories for state and routing
...
+`cluster.remote_store.routing_table.path.prefix` | "" (Empty string) | The fixed prefix to add for the index routing files in the blob store.
```
Comment (natebower) [inline _tuning-your-cluster/availability-and-recovery/remote-store/remote-cluster-state.md:102]: This and the next two lines appear to be duplicated content.

**Review by natebower — COMMENTED** [review]
@soosinha @Naarcha-AWS Please see my comments and let me know if you have any questions. Thanks!

File: _tuning-your-cluster/availability-and-recovery/remote-store/remote-cluster-state.md (hunk 1 above)
Comment (Naarcha-AWS) [inline _tuning-your-cluster/availability-and-recovery/remote-store/remote-cluster-state.md:99]:
```suggestion
`cluster.remote_store.state.path.prefix` | "" (Empty string) | The fixed prefix to add to the index metadata files in the blob store.
```

File: _tuning-your-cluster/availability-and-recovery/remote-store/remote-cluster-state.md (hunk 2 above)
Comment (Naarcha-AWS) [inline _tuning-your-cluster/availability-and-recovery/remote-store/remote-cluster-state.md:101]:
```suggestion
`cluster.remote_store.index_metadata.path_hash_algo` | `FNV_1A_BASE64 `  | The algorithm that constructs the prefix or infix for the index metadata path in the blob store. This setting is applied if the ``cluster.remote_store.index_metadata.path_type` setting is `HASHED_PREFIX` or `HASHED_INFIX`. Valid algorithm values are `FNV_1A_BASE64` and `FNV_1A_COMPOSITE_1`.
```

**Pull request merged**